web/src/message_view.js

```diff
@@ -299,7 +331,14 @@ export function show(raw_terms, opts) {
     if (raw_terms.length === 0) {
         raw_terms = [{operator: "is", operand: "home"}];
     }
-    const filter = new Filter(raw_terms);
+
+    // raw_terms with both `dm` and with operators are redundant, since dms
+    // don't have topics. Hence, we remove the `with` operator.
+    if (raw_terms.some((term) => term.operator === "dm")) {
+        raw_terms = raw_terms.filter((term) => term.operator !== "with");
+    }
```

Comment:
Should this be part of `Filter.fix_terms` ?

Reply:
oh ya, right! I'll update it.